langchain_tools_demo/tools.py

```diff
@@ -106,6 +106,37 @@ async def get_airport(id: int):
     return await response.json()
 
 
+class AirportSearchInput(BaseModel):
+    country: Optional[str] = Field(description="Country")
+    city: Optional[str] = Field(description="City")
+    name: Optional[str] = Field(description="Airport name")
+
+
+async def search_airports(country: str, city: str, name: str):
+    response = await get_request(
+        f"{BASE_URL}/airports/search",
+        {
+            "country": country,
+            "city": city,
+            "name": name,
+        },
+    )
+    if response.status != 200:
+        return f"Error searching airports: {response}"
+
+    num = 2
```

Comment:
nit: Do we need to truncate the response for airports? For flights it can list many results, but I don't expect the same issue here. The agent seems to struggle with 5-10+.

Reply:
Yes, user only have to provide at least one of (country, city, or name). So if the user only query with one criteria, it's very likely the database return 100s of rows.